### ISSUE TITLE
"Settings → Theme → System Theme" doesn't follow system theme

### ISSUE THREAD

**Issue by igalic** [flathub/codes.merritt.FeelingFinder#3]
When setting the Theme to System Theme, the theme is white no matter what the system theme currently is.

My OS is elementary OS 6.1 Jólnir.

**Comment by Merrit** [comment]
Hi @igalic :wave: 

Thanks for opening the issue!

This seems to be an issue with Flatpak being able to detect the system theme. I will take a look to see if there is something I can do as a workaround.

**Comment by Merrit** [comment]
It seems that flatpak & freedesktop have added a dbus item for [checking dark mode preference.](https://github.com/flatpak/xdg-desktop-portal/blob/d7a304a00697d7d608821253cd013f3b97ac0fb6/data/org.freedesktop.impl.portal.Settings.xml#L33-L45)

Since this is so new it isn't available on most systems yet. It should be available with GNOME 42+ and KDE 5.24+, looks like it will be available on Ubuntu 22.04.

Once it becomes available for me to test against I will look into implementing this check for the Flatpak build - probably soon, as it should be available to me in a couple of weeks.

***

**Notes:**

[Spec](https://flatpak.github.io/xdg-desktop-portal/#gdbus-org.freedesktop.portal.Settings):

> Portal API Reference
> 
> Portal interfaces are available to sandboxed applications with the default filtered session bus access of Flatpak. Unless otherwise specified, they appear under the bus name org.freedesktop.portal.Desktop and the object path /org/freedesktop/portal/desktop on the session bus. 
> 
> This interface provides read-only access to a small number of host settings required for toolkits similar to XSettings. It is not for general purpose settings.
> 
> Currently the interface provides the following keys:
> 
> org.freedesktop.appearance color-scheme u

>     Indicates the system's preferred color scheme. Supported values are:
>     0: No preference
>     1: Prefer dark appearance
>     2: Prefer light appearance
> 
>     Unknown values should be treated as 0 (no preference). 


Can be checked from terminal with:

```
dbus-send --print-reply --dest=org.freedesktop.portal.Desktop /org/freedesktop/portal/desktop org.freedesktop.portal.Settings.Read string:'org.freedesktop.appearance' string:'color-scheme'
```

**Comment by igalic** [comment]
Good evening.
Here's the current value:

```
meena@76ix ~> dbus-send --print-reply --dest=org.freedesktop.portal.Desktop /org/freedesktop/portal/desktop org.freedesktop.portal.Settings.Read string:'org.freedesktop.appearance' string:'color-scheme'
method return time=1649882829.950017 sender=:1.54 -> destination=:1.170 serial=292 reply_serial=2
   variant       variant          uint32 1
```

and after switching to light mode

```
meena@76ix ~> dbus-send --print-reply --dest=org.freedesktop.portal.Desktop /org/freedesktop/portal/desktop org.freedesktop.portal.Settings.Read string:'org.freedesktop.appearance' string:'color-scheme'
method return time=1649882860.108050 sender=:1.54 -> destination=:1.175 serial=332 reply_serial=2
   variant       variant          uint32 0
```

and switching back we get 1 again.

i have Flatpak 1.12.7 on this system.

**Comment by Merrit** [comment]
@igalic 

I implemented a check for the system theme where supported.
It will be available in v0.7.0, which should be pushing up to Flathub shortly.

**Comment by proninyaroslav** [comment]
@Merrit 
I found some problems with the system theme determining in Flatpak https://github.com/Merrit/flutter_flatpak/issues/1, this is related to your fix https://github.com/Merrit/feeling_finder/commit/c45b99f3acaeedad489c99bb47c24e5f723287c7
